cmd/agent/api/security.go

```diff
@@ -0,0 +1,114 @@
+package api
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/pkg/api/security"
+	"github.com/DataDog/datadog-agent/pkg/api/util"
+	grpc_auth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type contextKey int
+
+const (
+	contextKeyTokenInfoID contextKey = iota
+)
+
+var (
+	tlsKeyPair  *tls.Certificate
+	tlsCertPool *x509.CertPool
+	tlsAddr     string
+)
+
+// validateToken - validates token for legacy API
+func validateToken(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if err := util.Validate(w, r); err != nil {
+			return
```

Comment:
Would it be of any use to `WriteHeader(403)` here? probably not just a random thought

Reply:
I actually do think this makes sense.